diff --git a/src/common/logging/log.cpp b/src/common/logging/log.cpp
index b8d5cfe..448e0a1 100644
--- a/src/common/logging/log.cpp
+++ b/src/common/logging/log.cpp
@@ -211,23 +211,31 @@ void Setup(std::string_view log_filename) {
     spdlog::level default_log_level = spdlog::level::info;
     std::unordered_map<std::string, spdlog::level> log_level_per_class;
 
+    const auto range_to_string = [](const auto& range) {
+        std::string result;
+        for (const char c : range) {
+            result.push_back(c);
+        }
+        return result;
+    };
+
     if (EmulatorSettings.IsLogEnable()) {
         for (const auto class_level : std::views::split(EmulatorSettings.GetLogFilter(), ' ')) {
-            const auto class_level_pair =
-                std::views::split(class_level, ':') | std::ranges::to<std::vector<std::string>>();
+            std::vector<std::string> class_level_pair;
+            for (const auto part : std::views::split(class_level, ':')) {
+                class_level_pair.emplace_back(range_to_string(part));
+            }
 
             if (class_level_pair.size() != 2) {
                 std::cerr << "bad log filter provided" << std::endl;
                 continue;
             }
 
             if (class_level_pair.front()[0] == '*') {
-                default_log_level = spdlog::level_from_str(class_level_pair.back() |
-                                                           std::ranges::to<std::string>());
+                default_log_level = spdlog::level_from_str(class_level_pair.back());
             } else {
-                log_level_per_class[class_level_pair.front() | std::ranges::to<std::string>()] =
-                    spdlog::level_from_str(class_level_pair.back() |
-                                           std::ranges::to<std::string>());
+                log_level_per_class[class_level_pair.front()] =
+                    spdlog::level_from_str(class_level_pair.back());
             }
         }
     }
