diff --git a/src/osdep/amiberry.cpp b/src/osdep/amiberry.cpp
--- a/src/osdep/amiberry.cpp
+++ b/src/osdep/amiberry.cpp
@@ -7396 +7396 @@ static void resolve_bootstrap_settings_paths(const bool portable_mode, const bo
-	amiberry_ini_file = join_path(settings_dir, "amiberry.ini");
+	amiberry_ini_file = "/userdata/system/configs/amiberry/conf/amiberry.ini";
@@ -7398 +7398 @@ static void resolve_bootstrap_settings_paths(const bool portable_mode, const bo
-		amiberry_conf_file = join_path(settings_dir, "amiberry.conf");
+		amiberry_conf_file = "/userdata/system/configs/amiberry/conf/amiberry.conf";
