diff --git a/src/qt/CMakeLists.txt b/src/qt/CMakeLists.txt
index b3506b8..9fe97e7 100644
--- a/src/qt/CMakeLists.txt
+++ b/src/qt/CMakeLists.txt
@@ -214,5 +213,0 @@ add_library(ui STATIC
-    qt_vulkanwindowrenderer.hpp
-    qt_vulkanwindowrenderer.cpp
-
-    qt_vulkanrenderer.hpp
-    qt_vulkanrenderer.cpp
diff --git a/src/qt/qt_mainwindow.cpp b/src/qt/qt_mainwindow.cpp
index 3f915b8..d783afa 100644
--- a/src/qt/qt_mainwindow.cpp
+++ b/src/qt/qt_mainwindow.cpp
@@ -507 +507 @@ MainWindow::MainWindow(QWidget *parent)
-        if ((vid_api == RENDERER_OPENGL3) || (vid_api == RENDERER_VULKAN))
+        if (vid_api == RENDERER_OPENGL3)
@@ -520,15 +519,0 @@ MainWindow::MainWindow(QWidget *parent)
-#if QT_CONFIG(vulkan)
-    bool vulkanAvailable = false;
-    {
-        QVulkanInstance instance;
-        instance.setApiVersion(QVersionNumber(1, 0));
-        if (instance.create()) {
-            uint32_t physicalDevices = 0;
-            instance.functions()->vkEnumeratePhysicalDevices(instance.vkInstance(), &physicalDevices, nullptr);
-            if (physicalDevices != 0) {
-                vulkanAvailable = true;
-            }
-        }
-    }
-    if (!vulkanAvailable)
-#endif
@@ -544 +528,0 @@ MainWindow::MainWindow(QWidget *parent)
-    actGroup->addAction(ui->actionVulkan);
@@ -569,3 +552,0 @@ MainWindow::MainWindow(QWidget *parent)
-            case RENDERER_VULKAN:
-                newVidApi = RendererStack::Renderer::Vulkan;
-                break;
diff --git a/src/qt/qt_rendererstack.cpp b/src/qt/qt_rendererstack.cpp
index 9443cb8..f0e568f 100644
--- a/src/qt/qt_rendererstack.cpp
+++ b/src/qt/qt_rendererstack.cpp
@@ -24 +23,0 @@
-#include "qt_vulkanwindowrenderer.hpp"
@@ -419,36 +417,0 @@ RendererStack::createRenderer(Renderer renderer)
-#if QT_CONFIG(vulkan)
-        case Renderer::Vulkan:
-            {
-                this->createWinId();
-                VulkanWindowRenderer *hw = nullptr;
-                try {
-                    hw = new VulkanWindowRenderer(this);
-                } catch (std::runtime_error &e) {
-                    auto msgBox = new QMessageBox(QMessageBox::Critical, QString(), tr("Failed to initialize Vulkan renderer.") + QStringLiteral("\n") + e.what() + QStringLiteral("\n") + tr("Falling back to software rendering."), QMessageBox::Ok);
-                    msgBox->setAttribute(Qt::WA_DeleteOnClose);
-                    msgBox->show();
-                    imagebufs = {};
-                    QTimer::singleShot(0, this, [this]() { switchRenderer(Renderer::Software); });
-                    current.reset(nullptr);
-                    break;
-                }
-                rendererWindow = hw;
-                connect(this, &RendererStack::blitToRenderer, hw, &VulkanWindowRenderer::onBlit, Qt::QueuedConnection);
-                connect(hw, &VulkanWindowRenderer::rendererInitialized, [=]() {
-                    /* Buffers are available only after initialization. */
-                    imagebufs        = rendererWindow->getBuffers();
-                    switchInProgress = false;
-                    emit rendererChanged();
-                });
-                connect(hw, &VulkanWindowRenderer::errorInitializing, [=]() {
-                    /* Renderer could not initialize, fallback to software. */
-                    auto msgBox = new QMessageBox(QMessageBox::Critical, QString(), tr("Failed to initialize Vulkan renderer.") % QStringLiteral("\n") % tr("Falling back to software rendering."), QMessageBox::Ok);
-                    msgBox->setAttribute(Qt::WA_DeleteOnClose);
-                    msgBox->show();
-                    imagebufs = {};
-                    QTimer::singleShot(0, this, [this]() { switchRenderer(Renderer::Software); });
-                });
-                current.reset(this->createWindowContainer(hw));
-                break;
-            }
-#endif
@@ -472 +435 @@ RendererStack::createRenderer(Renderer renderer)
-    if (renderer != Renderer::OpenGL3 && renderer != Renderer::Vulkan) {
+    if (renderer != Renderer::OpenGL3) {
diff --git a/src/qt/qt_rendererstack.hpp b/src/qt/qt_rendererstack.hpp
index 02bf85a..ec02220 100644
--- a/src/qt/qt_rendererstack.hpp
+++ b/src/qt/qt_rendererstack.hpp
@@ -86 +85,0 @@ public:
-        Vulkan,
