diff --git a/include/input/cursor.h b/include/input/cursor.h
index 3d7aa71..fff2f4c 100644
--- a/include/input/cursor.h
+++ b/include/input/cursor.h
@@ -73,6 +73,8 @@ void cursor_set(struct seat *seat, enum lab_cursors cursor);
 
 void cursor_set_visible(struct seat *seat, bool visible);
 
+void cursor_set_permanent_hide(struct seat *seat, bool enable);
+
 /*
  * Safely store a cursor context to saved_ctx. saved_ctx is cleared when either
  * of its node, surface and view is destroyed.
diff --git a/include/labwc.h b/include/labwc.h
index 87a4219..88392cf 100644
--- a/include/labwc.h
+++ b/include/labwc.h
@@ -37,0 +38 @@ struct seat {
+	bool cursor_force_hidden;
diff --git a/src/action.c b/src/action.c
index 265ff6e..1bf885c 100644
--- a/src/action.c
+++ b/src/action.c
@@ -141,0 +142 @@ struct action_arg_list {
+	X(SHOW_CURSOR, "ShowCursor") \
@@ -1598,0 +1600,3 @@ run_action(struct view *view, struct action *action,
+	case ACTION_TYPE_SHOW_CURSOR:
+		cursor_set_permanent_hide(&server.seat, false);
+		break;
diff --git a/src/input/cursor.c b/src/input/cursor.c
index e8ac7d9..cd5743c 100644
--- a/src/input/cursor.c
+++ b/src/input/cursor.c
@@ -407,2 +407,9 @@ cursor_set_visible(struct seat *seat, bool visible)
 
+void
+cursor_set_permanent_hide(struct seat *seat, bool enable)
+{
+    seat->cursor_force_hidden = enable;
+    cursor_set_visible(seat, !enable);
+}
+
 void
