diff --git a/rpcs3/rpcs3qt/screenshot_manager_dialog.cpp b/rpcs3/rpcs3qt/screenshot_manager_dialog.cpp
--- a/rpcs3/rpcs3qt/screenshot_manager_dialog.cpp
+++ b/rpcs3/rpcs3qt/screenshot_manager_dialog.cpp
@@ -113 +113 @@ void screenshot_manager_dialog::reload()
-	const std::string screenshot_path_rpcs3 = fs::get_config_dir() + "screenshots/";
+	const std::string screenshot_path_rpcs3 = "/userdata/screenshots/";
